DataFormats/Detectors/EMCAL/include/DataFormatsEMCAL/AnalysisCluster.h

```diff
@@ -0,0 +1,227 @@
+                                               mCellIndex(cellIndex),
+                                               mMessage("Cell index " + std::to_string(mCellIndex) + " out of range.")
+    {
+    }
+
+    /// \brief Destructor
+    ~CellOutOfRangeException() noexcept final = default;
+
+    /// \brief Access to cell ID raising the exception
+    /// \return Cell ID
+    Int_t getCellIndex() const noexcept { return mCellIndex; }
+
+    /// \brief Access to error message of the exception
+    /// \return Error message
+    const char* what() const noexcept final { return mMessage.data(); }
+
+   private:
+    Int_t mCellIndex;     ///< Cell index raising the exception
+    std::string mMessage; ///< error Message
+  };
+
+  AnalysisCluster() = default;
+  ~AnalysisCluster() = default;
+  AnalysisCluster(const AnalysisCluster& clus) = default;
+  AnalysisCluster& operator=(const AnalysisCluster& source) = default;
+  void clear();
+
+  // Common EMCAL/PHOS/FMD/PMD
+
+  void setID(int id) { mID = id; }
+  int getID() const { return mID; }
+
+  void setE(double ene) { mEnergy = ene; }
+  double E() const { return mEnergy; }
+
+  void setChi2(double chi2) { mChi2 = chi2; }
+  double Chi2() const { return mChi2; }
+
+  ///
+  /// Set the cluster global position.
+  void setGlobalPosition(Point3D<double> x);
+  Point3D<double> getGlobalPosition() const
+  {
+    return mGlobalPos;
+  }
+
+  void setLocalPosition(Point3D<double> x);
+  Point3D<double> getLocalPosition() const
+  {
+    return mLocalPos;
+  }
+
+  void setDispersion(double disp) { mDispersion = disp; }
+  double getDispersion() const { return mDispersion; }
+
+  void setM20(double m20) { mM20 = m20; }
+  double getM20() const { return mM20; }
+
+  void setM02(double m02) { mM02 = m02; }
+  double getM02() const { return mM02; }
+
+  void setNExMax(unsigned char nExMax) { mNExMax = nExMax; }
+  unsigned char getNExMax() const { return mNExMax; }
+
+  void setEmcCpvDistance(double dEmcCpv) { mEmcCpvDistance = dEmcCpv; }
+  double getEmcCpvDistance() const { return mEmcCpvDistance; }
+  void setTrackDistance(double dx, double dz)
+  {
+    mTrackDx = dx;
+    mTrackDz = dz;
+  }
+  double getTrackDx() const { return mTrackDx; }
+  double getTrackDz() const { return mTrackDz; }
+
+  void setDistanceToBadChannel(double dist) { mDistToBadChannel = dist; }
+  double getDistanceToBadChannel() const { return mDistToBadChannel; }
+
+  void setNCells(int n) { mNCells = n; }
+  int getNCells() const { return mNCells; }
+
+  ///
+  ///  Set the array of cell indices.
+  void setCellsIndices(std::vector<unsigned short> array);
+  std::vector<unsigned short> getCellsIndices() { return mCellsIndices; }
+
+  ///
+  ///  Set the array of cell amplitude fractions.
+  ///  Cell can be shared between 2 clusters, here the fraction of energy
+  ///  assigned to each cluster is stored. Only in unfolded clusters.
+  void setCellsAmplitudeFraction(std::vector<Double32_t> array);
```

Comment:
the method is declared with vector argument, but in the cxx I see gsl::span. If you use a vector arg, use const ref instead of passing the copy. You can use also a span (it will accept the vector input also), but then, if you assign it to mCellsAmpFraction vector, you should take care of insertion, ``operator=`` will not work.